### ISSUE TITLE
Avoiding stackoverflow when dropping large List

### ISSUE THREAD

**Issue by lulitao1997** [orium/rpds#46]
As #41 suggests, when dropping a large List, rust recursively calls Arc's drop function, which leads to stackoverflow. We can implement our custom Drop trait for List to prevent this, the following code seems to solve this problem: 

```rust
impl<T, P> Drop for List<T, P>
where
    P: SharedPointerKind,
{
    fn drop(&mut self) {
        let mut head = self.head.take();
        while let Some(node) = head {
            if let Ok(mut node) = SharedPointer::try_unwrap(node) {
                head = node.next.take();
            }
            else {
                break;
            }
        }
    }
}
```

**Comment by lulitao1997** [comment]
can i make a pull request for this?

**Comment by orium** [comment]
Hi @lulitao1997.  A PR will be more than welcome :)

**Comment by orium** [comment]
Fixed by #47
